libs/src/api/serviceProvider.js

```diff
@@ -56,21 +56,23 @@ class ServiceProvider extends Base {
   }
 
   /**
-   * Fetches healthy creator nodes and autoselects a primary
-   * and two secondaries
-   * @param {number} numberOfNodes total number of nodes to fetch (2 secondaries means 3 total)
-   * @param {Set<string>?} whitelist whether or not to include only specified nodes (default no whiltelist)
-   * @param {Set<string?} blacklist whether or not to exclude any nodes (default no blacklist)
+   * Fetches healthy creator nodes, and then autoselects a primary and two secondaries
+   * @param {Object} param
+   * @param {number} param.[numberOfNodes=3] total number of nodes to fetch (3 = 1 primary, 2 secondaries)
+   * @param {Set<string>} param.[whitelist=null] whether or not to include only specified nodes
+   * @param {Set<string>} param.[blacklist=null]  whether or not to exclude any nodes
+   * @param {boolean} param.[performSyncCheck=true] flag to perform sync check or not
    * @returns { primary, secondaries, services }
    * // primary: string
-   * // secondaries: Array<string>
-   * // services: { creatorNodeEndpoint: versionInfo }
+   * // secondaries: string[]
+   * // services: { creatorNodeEndpoint: healthCheckResponse }
    */
-  async autoSelectCreatorNodes (
+  async autoSelectCreatorNodes ({
```

Comment:
good catch, just updated `sanityChecks:rolloverNodes` with the explicit parameters. 
in regards to removing the defaults, I think that maybe cause it is so sensitive that we would want to have defaults? this is a critical path, and having defaults ensures that we will never break here

Reply:
at least in regards to having values for the input params

Reply:
i don't agree - having defaults allows future code changes with reduced knowledge of function, and is prone to accidentally making the call with an incorrect value since it was defaulted. without defaults, every code change or addition will need to explicitly understand and define each param, which is much safer and less error prone.

but this is not a blocker for now, thanks for fixing the missing call